src/wazuh_modules/vulnerability_detector/wm_vuln_detector_evr.c

```diff
@@ -0,0 +1,470 @@
+        if (!*one) return -1;
+        if (!*two) return 1;
+        if (*one != '^') return 1;
+        if (*two != '^') return -1;
+        one++;
+        two++;
+        continue;
+    }
+
+    /* If we ran to the end of either, we are finished with the loop */
+    if (!(*one && *two)) break;
+
+    str1 = one;
+    str2 = two;
+
+    /* grab first completely alpha or completely numeric segment */
+    /* leave one and two pointing to the start of the alpha or numeric */
+    /* segment and walk str1 and str2 to end of segment */
+    if (c_isdigit(*str1)) {
+        while (*str1 && c_isdigit(*str1)) str1++;
+        while (*str2 && c_isdigit(*str2)) str2++;
+        isnum = 1;
+    } else {
+        while (*str1 && c_isalpha(*str1)) str1++;
+        while (*str2 && c_isalpha(*str2)) str2++;
+        isnum = 0;
+    }
+
+    /* save character at the end of the alpha or numeric segment */
+    /* so that they can be restored after the comparison */
+    oldch1 = *str1;
+    *str1 = '\0';
+    oldch2 = *str2;
+    *str2 = '\0';
+
+    /* this cannot happen, as we previously tested to make sure that */
+    /* the first string has a non-null segment */
+    if (one == str1) return -1;   /* arbitrary */
+
+    /* take care of the case where the two version segments are */
+    /* different types: one numeric, the other alpha (i.e. empty) */
+    /* numeric segments are always newer than alpha segments */
+    /* XXX See patch #60884 (and details) from bugzilla #50977. */
+    if (two == str2) return (isnum ? 1 : -1);
+
+    if (isnum) {
+        size_t onelen, twolen;
+        /* this used to be done by converting the digit segments */
+        /* to ints using atoi() - it's changed because long  */
+        /* digit segments can overflow an int - this should fix that. */
+
+        /* throw away any leading zeros - it's a number, right? */
+        while (*one == '0') one++;
+        while (*two == '0') two++;
+
+        /* whichever number has more digits wins */
+        onelen = strlen(one);
+        twolen = strlen(two);
+        if (onelen > twolen) return 1;
+        if (twolen > onelen) return -1;
+    }
+
+    /* strcmp will return which one is greater - even if the two */
+    /* segments are alpha or if they are numeric.  don't return  */
+    /* if they are equal because there might be more segments to */
+    /* compare */
+    rc = strcmp(one, two);
+    if (rc) return (rc < 1 ? -1 : 1);
+
+    /* restore character that was replaced by null above */
+    *str1 = oldch1;
+    one = str1;
+    *str2 = oldch2;
+    two = str2;
+    }
+
+    /* this catches the case where all numeric and alpha segments have */
+    /* compared identically but the segment sepparating characters were */
+    /* different */
+    if ((!*one) && (!*two)) return 0;
+
+    /* whichever version still has characters left over wins */
+    if (!*one) return -1; else return 1;
+}
+
+/**
+ * Compares two Debian versions. Extracted from Debian source code.
```

Comment:
It not used for Debian only after the latest changes, update the description.

Reply:
DONE